benchmarks/utils/ArborXBenchmark_PointClouds.hpp

```diff
@@ -49,160 +47,163 @@ inline PointCloudType to_point_cloud_enum(std::string const &str)
-      "The View should be accessible on the Host");
-  std::default_random_engine generator;
+  template <typename Random>
+  KOKKOS_FUNCTION auto hollowBoxPoint(int i, Random const &random) const
+  {
+    Point p;
+    // For 3D, the order is
+    // Indices: 0  1  2  3  4  5  6  7  8  9
+    // Axes   : 0- 0+ 1- 1+ 2- 2+ 0- 0+ 1- 1+
+    int axis = (i / 2) % DIM;
+    for (int d = 0; d < DIM; ++d)
+    {
+      if (d != axis)
+        p[d] = random();
+      else if (i % 2 == 0)
+        p[d] = -1;
+      else
+        p[d] = 1;
+    }
 
-  std::uniform_real_distribution<double> distribution(-radius, radius);
-  auto random = [&distribution, &generator]() {
-    return distribution(generator);
-  };
+    return p;
+  }
 
-  unsigned int const n = random_points.extent(0);
-  constexpr auto DIM = ArborX::GeometryTraits::dimension<Point>::value;
-  for (unsigned int i = 0; i < n; ++i)
+  template <typename Random>
+  KOKKOS_FUNCTION auto filledSpherePoint(Random const &random) const
   {
-    bool point_accepted = false;
-    while (!point_accepted)
+    Point p;
+    auto const radius = 1;
+    Coordinate norm;
+    do
     {
-      Point p;
-      double norm = 0.f;
+      norm = 0;
       for (int d = 0; d < DIM; ++d)
       {
-        double v = random();
-        p[d] = v;
-        norm += v * v;
+        p[d] = random();
+        norm += p[d] * p[d];
       }
-      norm = std::sqrt(norm);
+      norm = Kokkos::sqrt(norm);
 
       // Only accept points that are in the sphere
-      if (norm <= radius)
-      {
-        random_points(i) = p;
-        point_accepted = true;
-      }
-    }
-  }
-}
+    } while (norm > radius);
 
-template <class Point, typename... ViewProperties>
-void hollowSphereCloud(double const radius,
-                       Kokkos::View<Point *, ViewProperties...> random_points)
-{
-  namespace KokkosExt = ArborX::Details::KokkosExt;
-  static_assert(
-      KokkosExt::is_accessible_from_host<decltype(random_points)>::value,
-      "The View should be accessible on the Host");
-  std::default_random_engine generator;
-
-  std::normal_distribution<float> distribution(0.f, 1.f);
-  auto random = [&distribution, &generator]() {
-    return distribution(generator);
-  };
+    return p;
+  }
 
-  unsigned int const n = random_points.extent(0);
-  constexpr auto DIM = ArborX::GeometryTraits::dimension<Point>::value;
-  for (unsigned int i = 0; i < n; ++i)
+  template <typename Random>
+  KOKKOS_FUNCTION auto hollowSpherePoint(Random const &random) const
   {
-    double v[DIM];
-    double norm = 0.;
+    Point p;
+    Coordinate norm = 0;
     for (int d = 0; d < DIM; ++d)
     {
-      v[d] = random();
-      norm += v[d] * v[d];
+      p[d] = random();
+      norm += p[d] * p[d];
     }
-    norm = std::sqrt(norm);
-
+    norm = Kokkos::sqrt(norm);
     for (int d = 0; d < DIM; ++d)
-      random_points(i)[d] = radius * v[d] / norm;
+      p[d] /= norm;
```

Comment:
Maybe retry if we hit the origin?

Reply:
I'm changing this code as it does not generate uniformly distributed points on the sphere.